### PR TITLE
Make TagBuilder#attributes documentation public [ci-skip]

### PR DESCRIPTION
**Pull request by p8**
The `TagBuilder#attributes` seems to be [public AP](https://github.com/rails/rails/pull/40657)I but because TagBuilder is defined as `:nodoc` it doesn't show up in the API docs.

By adding `:doc:` for this method, it becomes visible in the API docs:
image

This PR makes only `attributes` public, but since more methods of `TagBuilder` are [described in `tag`](https://edgeapi.rubyonrails.org/classes/ActionView/Helpers/TagHelper.html#method-i-tag) maybe we can make more public?

### CODE REVIEW

File: actionview/lib/action_view/helpers/tag_helper.rb [hunk 1]
```diff
@@ -60,6 +61,7 @@ def initialize(view_context)
         def attributes(attributes)
```
Comment (eileencodes) [inline actionview/lib/action_view/helpers/tag_helper.rb:61]:
```suggestion
        def attributes(attributes) # :doc:
```

**Review by eileencodes — COMMENTED** [review]
From my search in the Rails codebase all you need is my code suggestion, the doc and nodoc around the method should be removed as they are already all nodoc'd.

File: actionview/lib/action_view/helpers/tag_helper.rb (hunk 1 above)
Comment (p8) [reply actionview/lib/action_view/helpers/tag_helper.rb:61]: Thanks @eileencodes.
I can't seem to make that work 😞. (Ruby 3.1, rdoc 6.4.0)
When I search for examples they all seem to document private methods of documented classes, not methods of `:nodoc:`-ed classes.
It seems like a `:nodoc:`-ed class should not be visible in the documentation even if a method is marked as `:doc:`, which sort of makes sense.

**Comment by p8** [comment]
Closing in favor of: https://github.com/rails/rails/pull/49368

**Pull request closed without merging**